zarp/plugins/sample_processors/defaults.py

```diff
@@ -35,7 +35,17 @@ def process(self) -> pd.DataFrame:
         LOGGER.info("Setting defaults...")
         LOGGER.warning("Plugin not implemented!")
         LOGGER.info("Defaults set: DESCRIBE WHAT WAS SET")
-        return self.records
 
-    def _select_records(self) -> None:
-        """Select records to process."""
+        sample_index = self.records.index
+
+        defaults = self.config.dict()
+        default_data = {}
+        for key, val in defaults:
+            default_data[key] = [val for _ in range(len(sample_index))]
+        defaults_df: pd.DataFrame(default_data)
+        default_df.index = sample_index
+
+        common_cols = default_df.columns.intersection(self.records.columns)
+        self.records = default_df.drop(common_cols, axis=1).join(self.records, how="right")
```

Comment:
If I read this correctly, this would only add columns that are not already in the sample table, not update any existing ones in the case that their values are `np.NaN`. Moreover, all columns that are required by ZARP are _already_ in the sample table, meaning that even if there were any columns added, ZARP wouldn't make use of them.

What we want instead is to update `self.records` with `default_df` in such a way that _only_ unset (`np.NaN`) values are set.

I would rather make use of `zarp.samples.sample_record_processor.SampleRecordProcessor.update()`:

```python
    def update(
        self,
        df: pd.DataFrame,
        by: Optional[str] = None,
        overwrite: bool = False,
        **kwargs: Any,
    ) -> None:
        """Update records with dataframe.

        Args:
            df: Pandas ``DataFrame`` object.
            by: Column to use as index for update. If ``None`` (default),
                compares database records by position.
            **kwargs: Keyword arguments to pass to ``_sanitize_df()``.

        Raises:
            KeyError: Column indicated by ``by`` not found in records or
                dataframe.
            ValueError: Records and dataframe have different lengths/rows.
        """
```

I'm just seeing that the description of the `overwrite` argument is missing. Its behavior is such that if set to `True` it will overwrite values regardless of their value. If set to `False` (default), it will only update missing (i.e., `np.NaN`) values, which is what we want here.

Beside the updating,  the `.update()` method also santizies the resulting data frame:
- It removes any columns _not_ in the original data frame
- It resolves any relative paths (or tries to, may not apply here; if it _does_ apply, you probably need to pass some `kwargs`)
- It replaces any `None` values with `np.NaN`
- It removes duplicates from the resulting table (by setting a hash sum compute over a row's values as the row name/index and then removing rows with identical row names)

You should be able to use it like this:

```python
from zarp.samples.sample_record_processor import SampleRecordProcessor as SRP

srp = SRP(self.records)
srp.update(df=default_df)

return srp.records
```

If the defaults contain paths, then you might need to further specify the `path_columns` and `anchor` arguents to `.update()` which would be passed along the this function (together with the data frame):

```python
def resolve_paths(
    df: pd.DataFrame,
    anchor: Path = Path.cwd(),
    path_columns: Sequence = tuple(columns_zarp_path),
) -> pd.DataFrame:
```